links/links_test.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package links
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"reflect"
+	"testing"
+
+	"golang.org/x/net/context"
+
+	"firebase.google.com/go/internal"
+	"google.golang.org/api/option"
+)
+
+var client *Client
+var testLinkStatsResponse []byte
+
+func TestMain(m *testing.M) {
+	defaultTestOpts := []option.ClientOption{
+		option.WithTokenSource(&internal.MockTokenSource{
+			AccessToken: "test-token",
+		})}
+
+	var err error
+	client, err = NewClient(context.Background(), defaultTestOpts...)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	testLinkStatsResponse, err = ioutil.ReadFile("../testdata/get_link_stats.json")
+	if err != nil {
+		log.Fatalln(err)
+	}
+	os.Exit(m.Run())
+}
+
+func TestCreateEventStatsMarshal(t *testing.T) {
+	es := &EventStats{Platform: Desktop, ET: AppFirstOpen, Count: 4}
+	m, err := json.Marshal(es)
+	if err != nil {
+		t.Error(err)
+	}
+	if string(m) != `{"platform":"DESKTOP","event":"APP_FIRST_OPEN","count":"4"}` {
+		t.Errorf(`Marshal(%v) = %v, expecting: {"platform":"DESKTOP","event":"APP_FIRST_OPEN","count":4}`,
+			es, string(m))
+	}
+}
+
+func TestCreateEventStatsString(t *testing.T) {
+	es := EventStats{Platform: IOS, ET: AppReOpen, Count: 4}
+
+	want := "{IOS APP_RE_OPEN 4}"
+	if str := fmt.Sprintf("%v", es); str != want {
+		t.Errorf("String representation of EventStats, got: %q, want: %q", str, want)
+	}
+}
+
+func TestReadJSON(t *testing.T) {
+	var ls LinkStats
+	err := json.Unmarshal(testLinkStatsResponse, &ls)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	want := getWantedStatResult()
+	if !reflect.DeepEqual(ls, want) {
+		t.Errorf("read json file, got %#v, want: %#v", ls, want)
```

Comment:
Please revise the error message format throughout this file. This should be: `LinkStats = %#v; want = %#v`